servicetalk-examples/grpc/compression/src/main/java/io/servicetalk/examples/grpc/compression/CompressionExampleServer.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.examples.grpc.compression;
+
+import io.servicetalk.concurrent.api.Single;
+import io.servicetalk.encoding.api.ContentCodec;
+import io.servicetalk.encoding.api.ContentCodings;
+import io.servicetalk.grpc.api.GrpcServiceContext;
+import io.servicetalk.grpc.netty.GrpcServers;
+import io.servicetalk.http.api.ContentCodingHttpServiceFilter;
+
+import io.grpc.examples.compression.Greeter.GreeterService;
+import io.grpc.examples.compression.Greeter.ServiceFactory;
+import io.grpc.examples.compression.HelloReply;
+import io.grpc.examples.compression.HelloRequest;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static io.servicetalk.concurrent.api.Single.succeeded;
+
+/**
+ * A simple extension of the gRPC "Hello World" example which demonstrates
+ * compression of the request and response bodies.
+ */
+public class CompressionExampleServer {
+
+    /**
+     * Supported encodings for the request. Requests using unsupported encodings receive
+     * an error response in the "grpc-status".
+     */
+    private static final List<ContentCodec> SUPPORTED_REQ_ENCODINGS =
```

Comment:
This list is not used. IIUC, because gRPC protocol (unlike HTTP) always expects compressed request messages from the client, there is no need to have two lists. Consider keeping only one list and removing `_REQ_/_RESP_` from the middle of the name: `SUPPORTED_ENCODINGS`.

Reply:
Correct. Now removed with suggested name change